pkg/server/plugin/keymanager/awskms/awskms.go

```diff
@@ -0,0 +1,533 @@
+			Type:     keyType,
+			PkixData: pub.PublicKey,
+		},
+	}, nil
+}
+
+func (p *Plugin) buildKeyEntry(ctx context.Context, aliasName *string, awsKeyID *string, spireKeyID string) (*keyEntry, error) {
+	describeResp, err := p.kmsClient.DescribeKey(ctx, &kms.DescribeKeyInput{KeyId: aliasName})
+	if err != nil {
+		return nil, kmsErr.New("failed to describe key: %v", err)
+	}
+
+	if !describeResp.KeyMetadata.Enabled {
+		// this means something external to the plugin, deleted or disabled the key without removing the alias
+		// returning an error provides the opportunity or reverting this in KMS
+		return nil, kmsErr.New("found disabled SPIRE key: %q, alias: %q", *awsKeyID, *aliasName)
+	}
+
+	keyType, err := keyTypeFromKeySpec(describeResp.KeyMetadata.CustomerMasterKeySpec)
+	if err != nil {
+		p.log.Debug("Skipped key", "reason", err)
+		return nil, nil
+	}
+
+	getPublicKeyResp, err := p.kmsClient.GetPublicKey(ctx, &kms.GetPublicKeyInput{KeyId: aliasName})
+	if err != nil {
+		return nil, kmsErr.New("failed to get public key: %v", err)
+	}
+
+	return &keyEntry{
+		KMSKeyID: *awsKeyID,
+		Alias:    *aliasName,
+		PublicKey: &keymanager.PublicKey{
+			Id:       spireKeyID,
+			Type:     keyType,
+			PkixData: getPublicKeyResp.PublicKey,
+		},
+	}, err
+}
+
+// fetchAliasesPage gets all aliases from KMS, and builds an in memory representation, with the ones that belong to the current server
+func (p *Plugin) fetchAliasesPage(ctx context.Context, marker *string) (*string, error) {
+	aliasesResp, err := p.kmsClient.ListAliases(ctx, &kms.ListAliasesInput{
+		Marker: marker,
+	})
+	if err != nil {
+		return nil, kmsErr.New("failed to fetch keys: %v", err)
+	}
+
+	p.log.Debug("Found aliases", "num_aliases", len(aliasesResp.Aliases))
+
+	for _, alias := range aliasesResp.Aliases {
+		if alias.AliasName == nil {
+			continue
+		}
+
+		spireKeyID, ok := p.spireKeyIDFromAlias(*alias.AliasName)
+		if !ok {
+			continue
+		}
+
+		if alias.TargetKeyId == nil {
+			// this means something external to the plugin created the alias, without associating it to a key.
+			// it should never happen with CMKs.
+			return nil, kmsErr.New("found SPIRE alias without key: %q", *alias.AliasName)
+		}
+
+		entry, err := p.buildKeyEntry(ctx, alias.AliasName, alias.TargetKeyId, spireKeyID)
+		switch {
+		case err != nil:
+			return nil, kmsErr.New("failed to process KMS key: %v", err)
+		case entry != nil:
+			err := p.setEntry(entry.PublicKey.Id, *entry)
+			if err != nil {
+				return nil, err
+			}
+			p.log.Debug("Key loaded", keyIDTag, *alias.TargetKeyId, aliasTag, *alias.AliasName)
+		}
+	}
+	return aliasesResp.NextMarker, nil
+}
+
+func (p *Plugin) scheduleDeleteTask(ctx context.Context) {
+	t := p.clock.Ticker(30 * time.Second)
+	defer t.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			p.log.Debug("Stopping schedule delete task", "reason", ctx.Err())
+			return
+		case <-t.C:
+			keyID, ok := <-p.scheduleDelete
+			if !ok {
+				continue
+			}
```

Comment:
Since the `key_id` log parameter is common to both success and error log messages, you can capture that in a local variable here to avoid repeating the inclusion of that parameter in both places:
```suggestion
			}
			log := p.log.With(keyIDTag, keyID)
```